.github/workflows/compile-deploy.yml

```diff
@@ -0,0 +1,27 @@
+# Compile, package and deploy the application on pushes to main, if tagged.
+name: compile-deploy
+
+on:
+  # Run for all pushes to main
+  push:
+    branches: [ main ]
+    tags: [ v* ]
+  # Allow the workflow to be run manually from the Actions tab
+  workflow_dispatch:
```

Comment:
I added @nathanklick to the review to look at the workflow. Do we have some easy way to reuse the basic workflow logic we're using in other repos? It would be good to have sonar and other things here

Reply:
@nathanklick would somebody from LimeChain be able to help out on this?

Reply:
I spoke with Nathan before the break. He said someone should be available to help with this portion after mid January or so.